### PR TITLE
feat: Settings 3000

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Inspired by [this](https://github.com/PostHog/posthog/issues/12923#issuecomment-1435104201) lets make settings way better.

1. Make it properly sectioned
2. Make it portable so we can embed product specific settings where they need to be
3. Clean up the components
4. Put it in the side panel
5. Remove as much antd as possible

## Changes

Screenshot 2023-11-06 at 17 10 27

Screenshot 2023-11-07 at 11 20 03

## TODO
- [x] Make sure it works with mobile
- [x] Fix hash params updating
- [x] Fix it for non-3000
- [x] Make SessionReplay modal use it

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/settings/project/SessionRecordingSettings.tsx [hunk 1]
```diff
@@ -1,30 +1,120 @@
...
+}
+
+export function ReplayCostControl(): JSX.Element {
     const { updateCurrentTeam } = useActions(teamLogic)
     const { currentTeam } = useValues(teamLogic)
 
     return (
         <FlaggedFeature flag={FEATURE_FLAGS.SESSION_RECORDING_SAMPLING}>
             <>
-                <h2 className={'flex flex-row align-center space-x-2'}>
```
Comment (daibhin) [inline frontend/src/scenes/settings/project/SessionRecordingSettings.tsx:25]: Kill?

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (daibhin) [inline .github/ISSUE_TEMPLATE/bug_report.md:1]: When I opened the settings menu in the sidebar I would have expected clicking the same button would close it again but seemed to only select the Project level
cannot-close

Comment (daibhin) [inline .github/ISSUE_TEMPLATE/bug_report.md:1]: Think this is beyond scope of your changes but the selected title style looks wrong here to me. I think it's the bolding with the light color. Goes away in 3000 so probably not worth a fix
Screenshot 2023-11-07 at 11 39 53

**Review by daibhin — APPROVED** [review]
There was _quite_ a lot here but I gave it my best review. I'm assuming most of the changes here were just moving files around.

Putting some individual things in separate threads

File: frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelSettings.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    const { selectedSectionId, selectedLevel } = useValues(settingsLogic(settingsLogicProps))
+
+    useEffect(() => {
+        setSettings({
+            sectionId: selectedSectionId ?? undefined,
+            settingLevelId: selectedLevel,
+        })
+    }, [selectedSectionId, selectedLevel])
+
+    // NOTE: Currently we can't detect url changes from the iframe
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/src/layout/navigation-3000/sidepanel/panels/SidePanelSettings.tsx:27]: Oh yeah

File: frontend/src/lib/components/JSBookmarklet.tsx [hunk 3]
```diff
@@ -10,12 +11,18 @@ export function JSBookmarklet({ team }: { team: TeamBasicType }): JSX.Element {
     )}%7D%7D)()`
 
     const { reportBookmarkletDragged } = useActions(eventUsageLogic)
+    const ref = useRef<HTMLAnchorElement>(null)
+
+    useEffect(() => {
+        // React cleverly stops js links from working, so we need to set the href manually
+        ref.current?.setAttribute('href', href)
+    }, [ref.current])
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/src/lib/components/JSBookmarklet.tsx:19]: Might as well make it clean 👍

File: frontend/src/scenes/settings/project/ProjectSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+    const { resetToken } = useActions(teamLogic)
+
+    return (
+        <div className="flex items-start gap-4 flex-wrap">
+            {/* <h2 id="custom-events" className="subtitle">
+                Send custom events
+            </h2>
+            To send custom events <Link to="https://posthog.com/docs/integrations">visit PostHog Docs</Link> and
+            integrate the library for the specific language or platform you're using. We support Python, Ruby, Node, Go,
+            PHP, iOS, Android, and more. */}
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/src/scenes/settings/project/ProjectSettings.tsx:172]: ah yeah this is all old

File: frontend/src/scenes/settings/project/SessionRecordingSettings.tsx (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/scenes/settings/project/SessionRecordingSettings.tsx:25]: yup -sorry I should have done a scan of this myself.

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (benjackwhite) [reply .github/ISSUE_TEMPLATE/bug_report.md:1]: Ah yeah I'm basing it on the change via useeffect... I'll see if there is a better way

Comment (benjackwhite) [reply .github/ISSUE_TEMPLATE/bug_report.md:1]: Yeah that was my thinking. Technically it works for that UI given that all buttons are that color.

Comment (benjackwhite) [reply .github/ISSUE_TEMPLATE/bug_report.md:1]: As always, moving the state into the logic is the key :D

File: frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx [hunk 5]
```diff
@@ -82,7 +82,8 @@ const ActivationSidebar = (): JSX.Element => {
                     />
                 </div>
                 {showSessionRecordingConfig ? (
-                    <SessionRecordingConfig />
+                    // TODO: FIX THIS
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx:85]: It does!

**Pull request merged**